### ISSUE TITLE
Location Report: make it so you can expand and collapse the location pop up

### ISSUE THREAD

**Issue by joshi1983** [hhaccessibility/hhaccessibility.github.io#244]
Test feedback showed people are annoyed with big portions of the map on the location report being covered by the question category and rating popup.  To address this problem, we want a way to toggle visibility of the popup.

Put an "X" in the popup that will close/hide the popup when clicked.

Make it so that when the marker is clicked and there is no popup, show the popup.

There is also a bug where the popup isn't vertically positioned to line up with the marker when the browser height is really tall or really short.  We'll want to fix that bug at some point too but fixing it as part of this issue is optional.

**Comment by joshi1983** [comment]
I assigned this to @e1cerebro because he worked on a few other issues related to Google Maps such as #221.  After #221 is complete, this may be a good fit for him.

**Comment by Blandine-AA** [comment]
> There is also a bug where the popup isn't vertically positioned to line up with the marker when the browser height is really tall or really short. We'll want to fix that bug at some point too but fixing it as part of this issue is optional.

I think it should be fixed at the same time. It seems more efficient that way.

**Comment by Blandine-AA** [comment]
@e1cerebro , before working on this issue, you should first complete the discovery and research on existing apps tasks.

**Comment by joshi1983** [comment]
I removed the following from the issue description because this bug has been fixed by the commits just before this comment:
"There is also a bug where the popup isn't vertically positioned to line up with the marker when the browser height is really tall or really short.  We'll want to fix that bug at some point too but fixing it as part of this issue is optional."

**Comment by joshi1983** [comment]
Work on this issue may affect work on #325 in that the toggle would work only when there is sufficient screen width for the pop up to look good on the map.

**Comment by Blandine-AA** [comment]
Un-assigning Chris since we haven't heard from him for a long time.
@joshi1983 , this will need to be the next priority. Can this be done for the 0.0.4 release?

**Comment by Blandine-AA** [comment]
Requirements for this issue has changed a little bit. See attached document.
[Location report web page changes.pdf](https://github.com/hhaccessibility/hhaccessibility.github.io/files/1383009/Location.report.web.page.changes.pdf)

**Comment by Blandine-AA** [comment]
Assigned to @princeseth to do the mockup based on the design description in the document attached in previous comment

**Comment by joshi1983** [comment]
It would be nice to have a mockup showing each of the following:
- mockup 1: how the location report page would look with the marker before it gets expanded.
- mockup 2: how the location report page would look after the marker is clicked
- mockup 3: how the location report page would look when user is not logged in and "Universal" is opened

**Comment by Blandine-AA** [comment]
@princeseth, @joshi1983 comments above are just to help you with the deliverables from the requirement/description document that I attached in this thread. 

Regarding the mobile/small screen issue, this was already reported in #325.

**Comment by princeseth** [comment]
@Blandine-AA @joshi1983 I have created three different mockups for **marker before it gets expanded**, **marker is clicked** and **not logged in and "Personal"**. Please see attached document
[Locatio_Mockup_0.4.pdf](https://github.com/hhaccessibility/hhaccessibility.github.io/files/1434874/Locatio_Mockup_0.4.pdf)

**Comment by Blandine-AA** [comment]
@princeseth, thanks for the effort on making the mockups.

Mockup number 2 is still not correct. The overall information section is not reflecting the design.
This is a side by side comparison
![image](https://user-images.githubusercontent.com/20052162/32290777-75954ff0-bf11-11e7-94c1-258a3948c8dd.png)
![image](https://user-images.githubusercontent.com/20052162/32291074-6272c640-bf12-11e7-9bca-52cc4e67d937.png)

The "(under development)" flag is not part of the mockup.

For the per category buttons box
- The is no need to have an (x) on the close button as you are not closing the button, but the window.
We can add an (X) at the top right of this window though.
![image](https://user-images.githubusercontent.com/20052162/32290961-0b362fca-bf12-11e7-8d3d-7da41fc402f1.png)

- For the pie chart, we need a simple one without the percentage info in it as it is redondant with the one display on the button.

Hope these comments help!

**Comment by princeseth** [comment]
@joshi1983 @Blandine-AA 
Please, have a look at below mockup 2 and share your feedback. Thanks.!
![mockup_2updated](https://user-images.githubusercontent.com/25910292/32343056-bae4d194-bfd8-11e7-91ac-f28ea1f4632b.png)

**Comment by Blandine-AA** [comment]
@princeseth, It looks much better. Just note that the pie chart has to be the same type for the overall section and the categories' buttons.

**Comment by joshi1983** [comment]
The mock ups look pretty good to me.  I like that the "x" is consistently in the top right corner.  I guess "Close" will be an alternative option in the lower centre that also can be clicked to close the popups.  I'll keep quiet as the discussion continues.

**Comment by Blandine-AA** [comment]
@joshi1983 , @princeseth, implementation can start on this one as I the other comments are more clarifications.
Please ensure that functionalities specifications are implemented as per the design document I provided.

**Comment by joshi1983** [comment]
That sounds good.  

Prince finished a little git-related task(making a pull request) quite well except that he used the github interface for it.

Prince, make a pull request as soon as possible with any trivial code change that you make locally using git commands.  Ideally, your implementation is documented with clear git commits so it is better to get the practice with git done before implementation.

**Comment by joshi1983** [comment]
The ordering of question categories is complete for this and reviewable at a place like http://app.accesslocator.com/location-report/387.  That affected the location report, rating, and user profile.

**Comment by joshi1983** [comment]
@princeseth please give a list of the remaining work on this issue and indicate which items you have working from the items not working.  I know you mentioned struggling with some CSS changes but do the JavaScript changes work and do they take steps toward looking more like the mockups?  

We can use the list to scope out a new pull request on this issue that you can do fairly independently.

**Comment by princeseth** [comment]
@joshi1983 I have achieved the functionality as per mockups, but struggling with some of the CSS changes.    
As of now, the remaining task is to fix CSS for blue box only.

**Comment by joshi1983** [comment]
@princeseth I asked for a list but that comment doesn't give one.  I want it to be very specific because I'll have to look for specific things when reviewing and know what items are remaining out of your pull request that may be completed in another pull request potentially by someone else.  I doubt you'll add the "x" button with CSS so there is at least one non-CSS item remaining.  You got the x button to work, right?

**Comment by princeseth** [comment]
@joshi1983 
Below image is similar to the mockups
![image](https://user-images.githubusercontent.com/25910292/33279582-0bd3551a-d36d-11e7-9f46-8d0667f056cc.png)

On clicking "x" button a map will expand and blue box will get close
![image](https://user-images.githubusercontent.com/25910292/33279830-ff504ba8-d36d-11e7-85a1-ec990634e09c.png)

On clicking the marker the blue box will appear again and a map will contract to an earlier position.

**Comment by princeseth** [comment]
@joshi1983 
 I`m struggling with the CSS of the blue box content marked in below image, when the name of the place exceeds from a single line.
![image](https://user-images.githubusercontent.com/25910292/33280796-47002b3c-d371-11e7-9f59-a2dbe01880ff.png)

**Comment by joshi1983** [comment]
I wanted to get a comprehensive list of items like mentioned below.  This is related to ability to break down problems into smaller sub-problems mentioned in a recent VIP pre-evaluation meeting.

Make a pull request with the following in it and we'll work through any challenges it may create first:
-[ ] Remove the arrow in the popup box for when rating percentages show
-[ ] Remove the arrow in Personal view when the explanation appears
-[ ] Align popup to the upper right of the Google map
One change from what you have in the screenshots is that I want the popup overlapping the map.  There won't be a wasteful gap below the popup box if the map is behind it.

The following can be handled in a later pull request after the above changes get merged:
-[ ] Have popup hidden upon initial load of the location report page
-[ ] Upon clicking the marker, pop up the expanded form
-[ ] Add a Close(X) in upper right corner of popup box
-[ ] When "X" button is clicked, hide the popup box.
-[ ] Add number of ratings to popup box
-[ ] Add number of comments to popup box

Are there other items that should be added to the list?

**Comment by princeseth** [comment]
I agree with you regarding popup overlapping the map, but the initial requirement was to open this info window beside the map, not on the top of the map. Blandine mentioned this in the document she provided to me.

In addition to the above list 
- [ ] comments (#) in the **title box** is a dynamic which will open a window with all the comments for a particular location

**Comment by joshi1983** [comment]
Can you point out the requirement in Blandine's document that says not to overlap the map apposed to just overlapping the right side of it?

It seems unclear to me from the PDF and the blue from the map is showing through which is a weak suggestion that it is fine to overlap the map:
![image](https://user-images.githubusercontent.com/11838191/33283391-9fe15926-d379-11e7-8a95-39af38686ab9.png)

**Comment by princeseth** [comment]
The third point in the document, she mentioned that info box on the side of the map "not" top.
![image](https://user-images.githubusercontent.com/25910292/33283736-bf4e5a56-d37a-11e7-9b8d-89a4a2f094cd.png)

You are correct, it is unclear from picture about overlapping the map which creates confusion. But, in the description, it is written not to overlap the map.

**Comment by joshi1983** [comment]
@Blandine-AA I suggest we overlap the right side of the map so the ugly gap below the popup box can be filled with the map's zoom + and - features and part of the map.  Is that ok?

**Comment by Blandine-AA** [comment]
I was thinking that the zoom feature would be on the side of the map as it is now. Also, there shouldn't be any gap as the Accessibility info Window will have the same height as the map. Have a look at this [Google map](https://www.google.ca/maps/place/Polytech+Lyon/@45.7780135,4.8488028,14z/data=!4m5!3m4!1s0x0:0xa34cbe87f6c6b03b!8m2!3d45.7793018!4d4.8683623) where the location info box is on the left.  They have the arrow to close and open the section.

**Comment by Blandine-AA** [comment]
I had a look at @princeseth screen captures above:
1- There shouldn't be a white space below the Criteria categories. See my comment above.
2- the Close button is missing

**Comment by joshi1983** [comment]
Matching the height of the map in all cases without leaving an empty gap anywhere might be tricky.  Google Maps uses all the space on the side:
![image](https://user-images.githubusercontent.com/11838191/33316767-236ae85a-d403-11e7-8e62-d0373fc6c2ee.png)

Our popup box content is a lot smaller than that.

Which of the following looks best?

Overlapping right side of map:
![image](https://user-images.githubusercontent.com/11838191/33316389-ce76534e-d401-11e7-8898-9f54c9d18555.png)

On right side of map:
![image](https://user-images.githubusercontent.com/11838191/33316509-2d1a4a86-d402-11e7-8311-dad5cac172e4.png)

On right side with gap filled with popup box background colour:
![image](https://user-images.githubusercontent.com/11838191/33316640-9eb3d3ce-d402-11e7-9029-b037538d8441.png)

The close button is the little "X" in the upper right corner of Prince's screenshots.

**Comment by Blandine-AA** [comment]
Hi @joshi1983 , @princeseth 
Thanks for all the screenshots and feedback.
1- I need and "Close" button at the bottom of the window in addition to the 'X'. The 'X' is not enough, plus it is barely visible.
2- I think we need to re-arrange the layout of the overall box info as it is too crowded. Looking at the google map example, we could do something similar with the location photo.
3- Looking at the box again. It is too dark. That's why it doesn't look nice in any screenshots. I suggest to redesign with with a lighter shade of blue and the grey. For the size and positioning I would go with something similar to the google map example, but with the zoom feature at the bottom because we have ou info window on the right and not on the left. Is this possible?
4- It is a good idea to move the location details (address, url ...) down in the box.
5- Eventually, the buttons can be a bit larger and wider. I see on @princeseth screenshot that the % is too close to the #rating and #comments info. Also "Public Areas" got too close to the % info. either "Public Areas" is wrapped-up or the window is widened to adjust.

**Comment by joshi1983** [comment]
@Blandine-AA yes, I agree on every point there.  

We could go lighter by changing the background to a near white with dark text.  This would likely be good to get mocked up or screen shot a few colour schemes.

Maybe we should stop having the "X" button at all and just have a clearer "Close" button centered under the categories.  That would save some space near the top for long location names.

**Comment by joshi1983** [comment]
@Blandine-AA mentioned that this needs to be done before next demo.accesslocator.com release so this is very important.  I'll review the discussion and remaining work needed for this fairly soon.

**Comment by joshi1983** [comment]
This should be limited to the non-mobile layout.  When the question categories are stacked, this toggle feature shouldn't exist.

**Comment by Blandine-AA** [comment]
Toggle doen't work properly on desktop.
Clicking on the pin should open AND close the info box. The current implementation only allows to open the info box. You have to close using the close button.

**Comment by Blandine-AA** [comment]
@joshi1983, this is the issue that covers the changes required in the accessibility info box. You event commented on this and summarized the work to be done. Attached is the initial document I created. It has not been updated with the feedback from the discussions in this thread, but the mockup for the accessibility categories buttons and the overall accessibility section remain.
[Location report web page changes.pdf](https://github.com/hhaccessibility/hhaccessibility.github.io/files/1741955/Location.report.web.page.changes.pdf)

**Comment by Blandine-AA** [comment]
@joshi1983, the All Comment link in the overall section needs to be implemented in 0.0.5 as part of that feature if it is not too complicated for the time we have.
A user may just be interested in the comments provided for that specific location.  You want to show them in a separate popup window.

**Comment by joshi1983** [comment]
I thought so too.  I made a new issue(#515) so @remnantmisook could implement the commenting feature while I'd just do the part of adding the link to the popup when she's done.

**Comment by Blandine-AA** [comment]
@joshi1983, added P1 label for this to be fixed within the next week.

**Comment by joshi1983** [comment]
This:
![image](https://user-images.githubusercontent.com/11838191/37766817-72f8e88c-2d9e-11e8-9806-786887732822.png)

was converted to this:
![image](https://user-images.githubusercontent.com/11838191/37766832-8481b6ce-2d9e-11e8-8efe-cd61402df18e.png)

In the latest pull request.

**Comment by Blandine-AA** [comment]
@joshi1983, it looks much better. Some feedback:

- [ ] Change "Comments" to "See All Comments"
- [ ] Remove brackets around the # of ratings for all the sections
- [ ] Remove brackets around 's' for ratings
- [ ] # of comments missing for the categories buttons. do not use brackets and comments to be plural.

Question: What happens if the location name is very long? Will the window size adjust accordingly?

**Comment by joshi1983** [comment]
If the location name is longer than 30 characters, the text shrinks.  It doesn't make the popup larger because a larger popup would look more inconsistent than just smaller text.

The best way you can see the dynamics of it is by finding some long named locations and looking at them on app.accesslocator.com.

**Comment by Blandine-AA** [comment]
Moving this to 0.0.6 as there is still some work to do on it

**Comment by joshi1983** [comment]
We want the phone number on these reports, right?  Roughly 80% of locations have phone numbers.  We also collect phone numbers in the Create New Location form.

I think the phone number would fit best in the red marked area here:
![image](https://user-images.githubusercontent.com/11838191/40971895-0569896e-688d-11e8-8eab-cb0986abc5ba.png)
